dbbackup/storage/filesystem_storage.py

```diff
@@ -3,58 +3,37 @@
 """
 from __future__ import (absolute_import, division,
                         print_function, unicode_literals)
-import os
-from .base import BaseStorage, StorageError
+import warnings
 
-from dbbackup import settings
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
 
+from .. import settings as dbbackup_settings
+from .base import StorageError
+from .builtin_django import Storage as DjangoStorage
 
-################################
-#  Filesystem Storage Object
-################################
+STORAGE_PATH = 'django.core.files.storage.FileSystemStorage'
 
-class Storage(BaseStorage):
-    """ Filesystem API Storage. """
 
-    def __init__(self, server_name=None):
-        self._check_filesystem_errors()
+class Storage(DjangoStorage):
+    """Filesystem API Storage."""
+
+    def __init__(self, server_name=None, **options):
         self.name = 'Filesystem'
-        BaseStorage.__init__(self)
+        options['location'] = options.get('location', dbbackup_settings.BACKUP_DIRECTORY)
+        super(Storage, self).__init__(storage_path=STORAGE_PATH,
+                                      **options)
+        self._check_filesystem_errors(options)
 
-    def _check_filesystem_errors(self):
+    def _check_filesystem_errors(self, options):
         """ Check we have all the required settings defined. """
-        if not self.backup_dir:
+        if options.get('location') is None:
             raise StorageError('Filesystem storage requires DBBACKUP_BACKUP_DIRECTORY to be defined in settings.')
-
-    ###################################
-    #  DBBackup Storage Methods
-    ###################################
-    
-    @property
-    def backup_dir(self):
-        return settings.BACKUP_DIRECTORY
-
-    def delete_file(self, filepath):
-        """ Delete the specified filepath. """
-        os.unlink(filepath)
-
-    def list_directory(self):
```

Comment:
It seems that list_directory is called in dbrestore and that it's called `listdir` in the builtin storage. Maybe just implement a proxy method for this and call listdir?